tests/end_to_end/helpers.h

```diff
@@ -307,6 +307,36 @@ typedef class _test_xdp_helper
     }
 } test_xdp_helper_t;
 
+static uint64_t
+_xdp_context_create(const uint8_t* input_buffer, size_t input_buffer_length, void** context)
+{
+    xdp_md_t* xdp_context = reinterpret_cast<xdp_md_t*>(malloc(sizeof(xdp_md_t)));
+    if (xdp_context == nullptr) {
+        return 1;
+    }
+
+    xdp_context->data = (void*)input_buffer;
+    xdp_context->data_end = (void*)(input_buffer + input_buffer_length);
```

Comment:
This is the part where it looks like it treats the whole input buffer as data_in and ctx_in isn't supported.